spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
Well, that's maybe how `ProjectIdProvider` would fix the concern. That is absolutely what we have with the `RegionProvider` in the SC AWS. But yes, that is fine to inject `GcpProperties` in to target code and perform some logic. 
In the end the `GcpProperties` is a bean anyway!